dev/io.openliberty.cdi.4.0.internal_fat/fat/src/io/openliberty/cdi40/internal/fat/config/LegacyConfigTest.java

```diff
@@ -0,0 +1,91 @@
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+import com.ibm.websphere.simplicity.config.Cdi;
+import com.ibm.websphere.simplicity.config.ConfigElementList;
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.annotation.TestServlets;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import io.openliberty.cdi40.internal.fat.config.beansxml.AllBeansServlet;
+import io.openliberty.cdi40.internal.fat.config.beansxml.RequestScopedBean;
+import io.openliberty.cdi40.internal.fat.config.beansxml.UnannotatedBean;
+
+@RunWith(FATRunner.class)
+public class LegacyConfigTest {
+    public static final String SERVER_NAME = "CDI40Server";
+
+    public static final String LEGACY_EMPTY_BEANS_APP_NAME = "LagacyEmptyBeans";
+
+    @Server(SERVER_NAME)
+    @TestServlets({
+                    @TestServlet(servlet = AllBeansServlet.class, contextRoot = LEGACY_EMPTY_BEANS_APP_NAME)
+    })
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        WebArchive legacyEmptyBeansXMLWar = ShrinkWrap.create(WebArchive.class, LEGACY_EMPTY_BEANS_APP_NAME + ".war")
+                                                      .addClass(AllBeansServlet.class)
+                                                      .addClass(UnannotatedBean.class)
+                                                      .addClass(RequestScopedBean.class);
+        CDIArchiveHelper.addEmptyBeansXML(legacyEmptyBeansXMLWar);
+        ShrinkHelper.exportDropinAppToServer(server, legacyEmptyBeansXMLWar, DeployOptions.SERVER_ONLY);
+
+        ServerConfiguration config = server.getServerConfiguration();
+        ConfigElementList<Cdi> cdis = config.getCdi();
+        Cdi cdi = new Cdi();
+        cdi.setEmptyBeansXMLExplicitArchive(true);
+        cdis.add(cdi);
+
+        server.updateServerConfiguration(config);
+
+        server.startServer(true, false);
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer();
+
+        ServerConfiguration config = server.getServerConfiguration();
+        ConfigElementList<Cdi> cdis = config.getCdi();
+        cdis.clear();
+        server.updateServerConfiguration(config);
+    }
+
+    /*
+     * Testing that the warning is NOT output on CDI 4.0
+     */
+    @Test
+    public void testCdiEmptyBeansXMLExplicitArchiveWarning() throws Exception {
+        List<String> warningMessages = server.findStringsInLogs("CWOWB1016W: The attribute cdiEmptyBeansXMLExplicitArchive of element type cdi is only supported on CDI 4.0 and newer. The attribute will be ignored.");
+        assertEquals("Message CWOWB1016W was found when it should not have been", 0, warningMessages.size());
```

Comment:
Did you test other 2 warnings? I could not find the other tests.

Reply:
https://github.com/OpenLiberty/open-liberty/pull/20196/files#diff-62ad3c5ed7c8e15972d920414fc3bae589a6c1c2749b21736c8e73468bfff01c